Source/OpenTK/Platform/Egl/EglContext.cs

```diff
@@ -97,9 +95,7 @@ abstract class EglContext : EmbeddedGraphicsContext
                 window.CreateWindowSurface(config);
 
             int[] attrib_list = new int[] { Egl.CONTEXT_CLIENT_VERSION, major, Egl.NONE };
-            HandleAsEGLContext = Egl.CreateContext(window.Display, config, shared != null ? shared.HandleAsEGLContext : IntPtr.Zero, attrib_list);
-
-            MakeCurrent(window);
+            HandleAsEGLContext = Egl.CreateContext(window.Display, config, sharedContext != null ? (sharedContext as IGraphicsContextInternal).Context.Handle : IntPtr.Zero, attrib_list);
```

Comment:
The cast on SharedContext violates [Liskov](https://en.wikipedia.org/wiki/Liskov_substitution_principle).

As a quick fix for robustness, could you perform another null-check after the cast? The alternative path handling can be really simple, use your best judgement for that.